pkg/resources/antiaffinity.go

```diff
@@ -25,23 +25,56 @@ import (
 
 // HostnameAntiAffinity returns a simple Affinity rule to prevent* scheduling of same kind pods on the same node.
 // *if scheduling is not possible with this rule, it will be ignored.
-func HostnameAntiAffinity(app string, antiAffinity kubermaticv1.AntiAffinityType) *corev1.Affinity {
+func HostnameAntiAffinity(app string, antiAffinityType kubermaticv1.AntiAffinityType) *corev1.Affinity {
+	return antiAffinity(app, antiAffinityType, TopologyKeyHostname)
+}
+
+// FailureDomainZoneAntiAffinity ensures that same-kind pods are spread across different availability zones.
+func FailureDomainZoneAntiAffinity(app string, antiAffinityType kubermaticv1.AntiAffinityType) *corev1.Affinity {
+	return antiAffinity(app, antiAffinityType, TopologyKeyZone)
+}
+
+func MergeAffinities(a *corev1.Affinity, b *corev1.Affinity) *corev1.Affinity {
+	mergedAffinity := &corev1.Affinity{
+		PodAntiAffinity: &corev1.PodAntiAffinity{},
+	}
+	if a == nil && b == nil {
+		return mergedAffinity
+	}
+	if a == nil {
+		return b
+	}
+	if b == nil {
+		return a
+	}
+	mergedAffinity.PodAntiAffinity.RequiredDuringSchedulingIgnoredDuringExecution = append(
+		a.PodAntiAffinity.RequiredDuringSchedulingIgnoredDuringExecution,
+		b.PodAntiAffinity.RequiredDuringSchedulingIgnoredDuringExecution...,
+	)
+	mergedAffinity.PodAntiAffinity.PreferredDuringSchedulingIgnoredDuringExecution = append(
+		a.PodAntiAffinity.PreferredDuringSchedulingIgnoredDuringExecution,
+		b.PodAntiAffinity.PreferredDuringSchedulingIgnoredDuringExecution...,
+	)
```

Comment:
The linter complains about this, so maybe we just merge everything into `a` (i.e. we append everything to `a` instead of creating `mergedAffinity`)?